tutorials/general/t8_tutorial_search.cxx

```diff
@@ -190,27 +202,28 @@ t8_tutorial_search_query_callback (t8_forest_t forest, t8_locidx_t ltreeid, cons
   T8_ASSERT (particles_per_element != NULL);
   T8_ASSERT (query != NULL);
 
-  /* Test whether this particle is inside this element. */
-  particle_is_inside_element
-    = t8_forest_element_point_inside (forest, ltreeid, element, particle->coordinates, tolerance);
-  if (particle_is_inside_element) {
-    if (is_leaf) {
-      /* The particle is inside and this element is a leaf element.
+  /* Test whether the particles are inside this element. */
+  t8_forest_element_point_batch_inside (forest, ltreeid, element, coords, num_active_queries, query_matches, tolerance);
+  T8_FREE (coords);
+  for (size_t matches_id = 0; matches_id < num_active_queries; matches_id++) {
+    if (query_matches[matches_id]) {
+      if (is_leaf) {
+        /* The particle is inside and this element is a leaf element.
        * We mark the particle for being inside the partition and we increase
        * the particles_per_element counter of this element. */
-      /* In order to find the index of the element inside the array, we compute the
+        /* In order to find the index of the element inside the array, we compute the
        * index of the first element of this tree plus the index of the element within
        * the tree. */
-      t8_locidx_t element_index = t8_forest_get_tree_element_offset (forest, ltreeid) + tree_leaf_index;
-      particle->is_inside_partition = 1;
-      *(double *) t8_sc_array_index_locidx (particles_per_element, element_index) += 1;
+        t8_locidx_t element_index = t8_forest_get_tree_element_offset (forest, ltreeid) + tree_leaf_index;
+        /* Get the correct particle_id */
+        size_t particle_id = *(size_t *) sc_array_index_int (query_indices, matches_id);
+        t8_tutorial_search_particle_t *particle
+          = (t8_tutorial_search_particle_t *) sc_array_index ((sc_array_t *) query, particle_id);
+        particle->is_inside_partition = 1;
+        *(double *) t8_sc_array_index_locidx (particles_per_element, element_index) += 1;
+      }
     }
-    /* The particles is inside the element. This query should remain active.
-     * If this element is not a leaf the search will continue with its children. */
-    return 1;
```

Comment:
Is this logic not needed anymore?

Reply:
No, this logic has been moved into the search-function itself.